src/main/send_main.cc

```diff
@@ -4,14 +4,90 @@
 
 #include "src/IPC/IPC.h"
 
-int main()
+#include "ipc_common.h"
+
+int main(int argc, char* argv[])
 {
-    char path[] = {"/home/maksim/ipc-class/input.txt"};
-    std::fstream file;
-
-    std::cout << "App: Launch Pipe Tx" << std::endl;
-    CreatorIPC* pipe_tx = new CreatorPipeTx();
-    file = pipe_tx->openReadFile(path);
-    SenderIPC* pipe_file_tx = pipe_tx->createIpcTx(&file);
-    std::cout << pipe_file_tx->send();
+    int c;
+    Arguments arg;
+
+    /* Default arguments values */
+    arg.tr_type = DEFAULT;
+    arg.read_path = NULL;
+    arg.write_path = NULL;
+
+    /* if not enough arguments given */
+    if (argc < 2) {
+        std::cerr << "ERROR: send_main requires arguments \n" << std::endl;
+        return -1;
+    }
+
+    while (1) {
+        int option_index = 0;
+
+        c = getopt_long(argc, argv, "hqpsf:", long_options, &option_index);
+
+        /* detect the end of options */
+        if (c == -1)
+            break;
+        
+        switch (c) {
+            case 'q':
+                arg.tr_type = QUEUE;
+                break;
+            case 'p':
+                arg.tr_type = PIPE;
+                break;
+            case 's':
+                arg.tr_type = SHM;
+                break;
+            case 'f':
+            arg.read_path = optarg;
+            break;
+
+            case 'h':
+                std::cout << "\n Use the program as follows: \n" << std::endl;
+                std::cout << "--help: prints out a help text containing short description of all supported command-line arguments \n" << std::endl;
+                std::cout << "--queue <TBD>: use message queue as IPC method \n" << std::endl;
+                std::cout << "--pipe <TBD>: use pipe as IPC method \n" << std::endl;
+                std::cout << "--shm: use shared memory buffer as IPC method \n" << std::endl;
+                std::cout << "--file [filename]: file used to read data from \n" << std::endl;
+                return -1;
+            
+            default:
+                std::cerr << "ERROR: unknown arguments provided" << std::endl;
+                return -1;
+        }
+    }
+
+    if (!arg.read_path) {
+        std::cerr << "ERROR: no file path provided" << std::endl;
+        return -1;
+    }
+
+    switch (arg.tr_type) {
+        case PIPE: {
+            std::fstream file;
+            std::cout << "Launch Pipe Tx" << std::endl;
+            CreatorIPC* pipe_tx = new CreatorPipeTx();
+            file = pipe_tx->openReadFile(arg.read_path);
+            SenderIPC* pipe_file_tx = pipe_tx->createIpcTx(&file);
```

Comment:
You can consider using some smart pointers in 'createIpcTx' (perhaps unique_ptr). This will make life easier to api user, because, there will be clear indication who is the owner of object and who need to delete.
No strong opinion here, you can also just document your api, that user is responsible to delete ptr
And btw, will be good if file param will be passed with const, I guess you don't want to allow any changes of it here?